### PR TITLE
[Spec Decode] Make EAGLE3 draft token ID mapping optional

### PR DESCRIPTION
**Pull request by benchislett**
Small compatibility fix for EAGLE3 models which do not have the "d2t" mapping defined. In this case, the logits can be returned as-is.

### CODE REVIEW

File: vllm/v1/spec_decode/eagle.py [hunk 1]
```diff
@@ -334,6 +334,14 @@ def load_model(self, target_model: nn.Module) -> None:
             logger.info("Loading EAGLE LM head weights from the target model.")
             self.model.lm_head = target_model.lm_head
 
+        if self.vllm_config.speculative_config.method == "eagle3" and \
+                hasattr(target_model, "draft_id_to_target_id") and \
+                "d2t" not in loaded_weights:
+            logger.info("Draft id to target id mapping not found. Assuming "
+                        "draft and target model share the same vocab.")
+            del self.model.draft_id_to_target_id
+            self.model.draft_id_to_target_id = None
```
Comment (WoosukKwon) [inline vllm/v1/spec_decode/eagle.py:343]: Why not fix this in eagle3 model?

Comment (benchislett) [reply vllm/v1/spec_decode/eagle.py]: Good point. I hadn't noticed that all the other logic in `eagle.py` exists to copy data from the target model, which is not needed here. Changed.

File: vllm/model_executor/models/llama_eagle3.py [hunk 2]
```diff
@@ -245,4 +248,10 @@ def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]):
                 name = "model." + name
             model_weights[name] = loaded_weight
 
-        return loader.load_weights(model_weights.items())
+        loaded_weights = loader.load_weights(model_weights.items())
+
+        if 'd2t' not in loaded_weights:
+            del self.draft_id_to_target_id
+            self.draft_id_to_target_id = None
```
Comment (WoosukKwon) [inline vllm/model_executor/models/llama_eagle3.py:255]:
```suggestion
            self.draft_id_to_target_id = None
```

Comment (WoosukKwon) [reply vllm/model_executor/models/llama_eagle3.py:255]: why `del` here?

Comment (benchislett) [reply vllm/model_executor/models/llama_eagle3.py]: I thought it might clear the GPU memory immediately, but that seems not to be the case. Fixed.

**Review by WoosukKwon — APPROVED** [review]
LGTM! Thanks for the PR!
